### PR TITLE
feat: add UI template for maintenance

### PR DESCRIPTION
**Pull request by YaroslavOkrema**
image

### CODE REVIEW

**Review by Roman15122022 — CHANGES_REQUESTED** [review]
Use shadcn inputs and btns

And try to use all screen area

File: src/pages/ServiceMaintenance/index.tsx [hunk 1]
```diff
@@ -1,9 +1,97 @@
...
-      <div>ServiceMaintenance</div>
+      <div className="p-6 max-w-4xl mx-auto">
+        <Card>
+          <CardHeader>
+            <h1 className="text-xl font-semibold">Maintenance</h1>
+          </CardHeader>
+          <CardContent>
+            <form className="space-y-4">
+              <div>
+                <h1>Service Date</h1>
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/index.tsx:26]: use locale

File: src/pages/ServiceMaintenance/index.tsx [hunk 2]
```diff
@@ -1,9 +1,97 @@
...
+                  placeholder="Enter the address"
+                />
+              </div>
+              <div>
+                <h1>Odometer reading</h1>
+                <input
+                  className="text-black"
+                  id="odometer"
+                  type="number"
+                  placeholder="Current mileage"
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/index.tsx:66]: locale

File: src/pages/ServiceMaintenance/index.tsx [hunk 3]
```diff
@@ -1,9 +1,97 @@
...
+                />
+              </div>
+              <div>
+                <h1>Details of service</h1>
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/index.tsx:70]: same

File: src/pages/ServiceMaintenance/index.tsx [hunk 4]
```diff
@@ -1,9 +1,97 @@
...
+                <Textarea placeholder="Type your message here." />
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/index.tsx:71]: same

**Review by Roman15122022 — CHANGES_REQUESTED** [review]

**Comment by YaroslavOkrema** [comment]
image

File: src/pages/ServiceMaintenance/useMaintenance.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    odometerPlaceholder:
+      translations.maintenance.addMaintenance.odometerPlaceholder,
+    details: translations.maintenance.addMaintenance.details,
+    detailsPlaceholder:
+      translations.maintenance.addMaintenance.detailsPlaceholder,
+    cost: translations.maintenance.addMaintenance.cost,
+    costPlaceholder: translations.maintenance.addMaintenance.costPlaceholder,
+    button: translations.maintenance.addMaintenance.button,
+  }
+}
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/useMaintenance.ts:35]: just return
`locale: translations.maintenance.addMaintenance`

and use it in component
like:
locale.button
locale.cost etc

File: src/pages/ServiceMaintenance/index.tsx [hunk 6]
```diff
@@ -1,9 +1,141 @@
...
+    address,
+    addressPlaceholder,
+    odometer,
+    odometerPlaceholder,
+    details,
+    detailsPlaceholder,
+    cost,
+    costPlaceholder,
+    button,
+  } = useMaintenance()
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/index.tsx:51]: too many elements

File: src/pages/ServiceMaintenance/useMaintenance.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/useMaintenance.ts:1]: for what import React

File: src/pages/ServiceMaintenance/index.tsx [hunk 8]
```diff
@@ -1,9 +1,141 @@
+import React from 'react'
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/index.tsx:1]: for what import React

**Review by Roman15122022 — APPROVED** [review]
In general ok

File: src/pages/ServiceMaintenance/index.tsx [hunk 9]
```diff
@@ -1,9 +1,141 @@
...
+                  <Button
+                    variant="outline"
+                    className={cn(
+                      'w-[240px] justify-start text-left font-normal',
+                      !date && 'text-muted-foreground',
+                    )}
+                  >
+                    <CalendarIcon />
+                    {date ? (
+                      format(date, 'PPP')
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/index.tsx:74]: constant for format

File: src/pages/ServiceMaintenance/index.tsx [hunk 10]
```diff
@@ -1,9 +1,141 @@
...
+        <CardContent>
+          <form className="space-y-4">
+            <div>
+              <h1 className="text-lg font-medium mb-2">{serviceDate}</h1>
+              <Popover>
+                <PopoverTrigger asChild>
+                  <Button
+                    variant="outline"
+                    className={cn(
+                      'w-[240px] justify-start text-left font-normal',
```
Comment (Roman15122022) [inline src/pages/ServiceMaintenance/index.tsx:68]: rarely use this constuction [] - tailwind have wonderful size grid

**Review by Roman15122022 — APPROVED** [review]

**Pull request merged**